### PR TITLE
[Az.RecoveryServices.Backup] Fixed bugs, added documentation and examples

### PR DESCRIPTION
**Pull request by hiaga**
Fixed bug with rp expiry time, making 30 days expiry time for adhoc backup as default from client side. Added example to fetch pruned recovery points after modify policy. Fixed the documentation for suspend backups with immutability.

## Description
Regenerated CRR SDK. Fixed issues with SQL CRR.
Fixed bug with rp expiry time, making 30 days expiry time for adhoc backup as default from client side.
Added example to fetch pruned recovery points after modify policy.
Fixed the documentation for suspend backups with immutability.
Regenerated CRR SDK. Fixed issues with SQL CRR.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs [hunk 1]
```diff
@@ -105,6 +106,12 @@ public class GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig : RSBackupVa
         [Parameter(Mandatory = false, HelpMessage = ParamHelpMsgs.RecoveryPointConfig.FilePath)]
         public string FilePath { get; set; }
 
+        /// <summary>
+        /// Switch parameter to specify fetching resources from Secondary Region.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = ParamHelpMsgs.Common.UseSecondaryReg)]
+        public SwitchParameter UseSecondaryRegion;
```
Comment (vidai-msft) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs:113]: Please re-generated the help doc for this cmdlet due to syntax update.

**Review by vidai-msft — CHANGES_REQUESTED** [review]

**Comment by vidai-msft** [comment]
@hiaga can you please also add test cases to verify the code change?

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs (hunk 1 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs:113]: done, pls check

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs [hunk 2]
```diff
@@ -218,22 +225,44 @@ public override void ExecuteCmdlet()
                             ((AzureWorkloadProtectableItem)TargetItem).ServerName) == 0)
                         {
                             string itemId = GetItemId(RecoveryPoint.Id);
-                            IList<SQLDataDirectory> dataDirectoryPaths = GetRpDetails(vaultName, resourceGroupName);
-                            foreach (var dataDirectoryPath in dataDirectoryPaths)
+
+                            if(UseSecondaryRegion)
```
Comment (vidai-msft) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs:229]: Is it possible to add a test case for this cmdlet update?

Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs:229]: It is slightly tricky to add the test case for this. I'm thinking for the same. I'll let you know by EOD ? will it be ok ?

Comment (vidai-msft) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs:229]: It is OK. Thank you!

Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/GetAzureRmRecoveryServicesBackupWorkloadRecoveryConfig.cs:229]: I've added the test case, pls check now!

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Jobs/GetAzureRmRecoveryServicesBackupJob.cs [hunk 3]
```diff
@@ -167,8 +176,12 @@ public override void ExecuteCmdlet()
                 int resultCount = 0;
                 if (UseSecondaryRegion.IsPresent)
                 {
-                    ARSVault vault = ServiceClientAdapter.GetVault(resourceGroupName, vaultName);
-                    string secondaryRegion = BackupUtils.regionMap[vault.Location];
+                    if (VaultLocation == null || VaultLocation == "")
```
Comment (vidai-msft) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Jobs/GetAzureRmRecoveryServicesBackupJob.cs:179]: VaultLocation already had attribute [ValidateNotNullOrEmpty]. Could this if condition be true?

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Jobs/GetAzureRmRecoveryServicesBackupJobDetails.cs [hunk 4]
```diff
@@ -74,10 +84,12 @@ public override void ExecuteCmdlet()
                     CrrModel.CrrJobRequest jobRequest = new CrrModel.CrrJobRequest();
                     jobRequest.JobName = JobId;
                     jobRequest.ResourceId = VaultId;
+                                                            
+                    if(VaultLocation == null || VaultLocation == "") {
```
Comment (vidai-msft) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Jobs/GetAzureRmRecoveryServicesBackupJobDetails.cs:88]: Same question

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs [hunk 5]
```diff
@@ -338,8 +338,7 @@ public override void ExecuteCmdlet()
                     }
                     else
                     {
-                        ARSVault vault = ServiceClientAdapter.GetVault(resourceGroupName, vaultName);
-                        secondaryRegion = BackupUtils.regionMap[vault.Location];
+                        throw new PSArgumentException(Resources.VaultLocationRequiredForCRR);
```
Comment (vidai-msft) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs:341]: This piece of code looks like never reached.

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Jobs/GetAzureRmRecoveryServicesBackupJobDetails.cs (hunk 4 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Jobs/GetAzureRmRecoveryServicesBackupJobDetails.cs:88]: same

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs (hunk 5 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs:341]: this code may reach when RestoreToSecondaryRegion switch is given but VaultLocation isn't given.

**Review by vidai-msft — APPROVED** [review]

**Pull request merged**